gateway/src/resty/http/proxy.lua

```diff
@@ -83,9 +83,7 @@ local function connect_proxy(httpc, request, skip_https_connect)
         proxy_uri.port = default_port(proxy_uri)
     end
 
-    if not port then
-        port = default_port(uri)
-    end
+    local port = default_port(uri)
```

Comment:
no, it's not breaking existing behavior. 

`port` was not declared local, hence it was accessing some (not existing) global variable. Nginx was rightfully complaining about that because it could be causing race conditions. It seems me that there used to be some local `port` declared and it was removed. :shrug: